### PR TITLE
Support disable leader election for manager go binary via Values.yaml to mitigate kuberay restarts

### PR DESCRIPTION
**Pull request by aviadshimoni**
## Why are these changes needed?
Currently, we don't have support to disable leader election via kuberay values.yaml
to disable leader election and prevent leader election lost errors:
```
leaderElectionEnabled: false
```

true is the default, so there is no need to check if enabled

https://github.com/ray-project/kuberay/issues/2252
https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1717181822513479

## Related issue number
https://github.com/ray-project/kuberay/issues/2252

## Checks

- [ ] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by aviadshimoni** [comment]
Testing manually in stg:
image


will update the result and if restart persists

**Comment by jjyao** [comment]
cc @andrewsykim

**Comment by andrewsykim** [comment]
CI failure:
```
[2024-07-21T12:43:45Z] Error: INSTALLATION FAILED: template: kuberay-operator/templates/deployment.yaml:84:30: executing "kuberay-operator/templates/deployment.yaml" at <.Values.leaderElection.enabled>: nil pointer evaluating interface {}.enabled
```

File: helm-chart/kuberay-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -81,6 +81,9 @@ spec:
             {{- $argList = append $argList "--log-file-encoder" -}}
             {{- $argList = append $argList .Values.logging.fileEncoder -}}
             {{- end -}}
+            {{- if not .Values.leaderElection.enabled -}}
```
Comment (andrewsykim) [inline helm-chart/kuberay-operator/templates/deployment.yaml:84]: Define the default value in https://github.com/ray-project/kuberay/blob/master/helm-chart/kuberay-operator/values.yaml

**Comment by aviadshimoni** [comment]
image

seems that no restart occur when disabling leader election, to me, it seems like we hit k8s limit API for some reason.

@andrewsykim what do you think?
what is better? 
1. have restarts every 60mins but ability to grow to more replicas
2. disable leader election and have 1 replica

File: helm-chart/kuberay-operator/values.yaml [hunk 2]
```diff
@@ -74,6 +74,10 @@ securityContext:
     type: RuntimeDefault
 
 
+# enabled by default, disable leader election to solve k8s api timeout when leader election occur and 1 replica exist.
+leaderElection:
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/values.yaml:78]: Will there be any other options under `leaderElection`? If not, maybe we can use `leaderElectionEnabled` instead.

Comment (aviadshimoni) [reply helm-chart/kuberay-operator/values.yaml]: Thank for the suggestion,
@kevin85421 seems like we hit API server limit, the question is why?

Comment (aviadshimoni) [reply helm-chart/kuberay-operator/values.yaml]: should I have more than 1 replicas in kuberay? if so, I don't want to disable leader election.

**Comment by aviadshimoni** [comment]
Pipeline passed :)
added defaults

**Comment by kevin85421** [comment]
I plan to include this PR in KubeRay v1.2.0. The branch cut is targeted on this Friday.

**Pull request merged**